zaza/utilities/networking.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2018 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module for working with networking."""
+
+import ipaddress
+
+
+def format_addr(addr):
+    """Validate and format IP address.
+
+    :param addr: IPv6 or IPv4 address
+    :type addr: str
+    :returns: Address string, optionally encapsulated in brackets([])
+    :rtype: str
+    :raises: ValueError
+    """
+    ipaddr = ipaddress.ip_address(addr)
+    if isinstance(ipaddr, ipaddress.IPv6Address):
+        fmt = '[{}]'
+    else:
+        fmt = '{}'
+    return fmt.format(ipaddr)
```

Comment:
This is a little odd.  Why noe

```python
    if isinstance(ipaddr, ipaddress.IPv6Address):
        return '[{}]'.format(ipaddr)
    else:
        return str(ipaddr)
```

Reply:
Mostly because this change is just porting a function from [zaza-openstack-tests](https://github.com/openstack-charmers/zaza-openstack-tests/blob/29356f6419ef76dd182c63322bdba5edcfcefac9/zaza/openstack/charm_tests/test_utils.py#L546-L560)